wolfcrypt/src/pkcs7.c

```diff
@@ -4560,14 +4560,36 @@ static int PKCS7_VerifySignedData(PKCS7* pkcs7, const byte* hashBuf,
             /* Get the inner ContentInfo contentType */
             if (ret == 0) {
                 word32 tmpIdx = idx;
-
-                if (GetASNObjectId(pkiMsg, &idx, &length, pkiMsgSz) != 0)
-                    ret = ASN_PARSE_E;
-
-                contentType = pkiMsg + tmpIdx;
-                contentTypeSz = length + (idx - tmpIdx);
-
-                idx += length;
+                if (length == 0 && pkiMsg[idx-1] == ASN_INDEF_LENGTH) {
+                    ret = BER_INDEF_E;
+                #ifdef ASN_BER_TO_DER
+                    /* content type OID has indef encoding */
+                    /* locate ASN_EOC */
+                    while (tmpIdx + 1 < pkiMsgSz) {
```

Comment:
When the object is indefinite length then is constructed and has one or more not-constructed objects underneath.
Instead of searching for the end, look for another ASN.1 Object inside which is definite length.
Then the contents of the ASN.1 Object is the OID and the EOC is after.
(If there were to be multiple ASN.1 Objects underneath then they would have to be coalesced.)

Reply:
There is no length in this real life example from our customer. The OID value is indef encoded with 0x80 leading and 0x00, 0x00 trailing. I cannot determine length by any means. See the ZD ticket for the PKCS7 certificate.

Reply:
The ASN.1 OBJECT is not indefinite length encoded in cacerts.p7c.
The SEQUENCE is indefinite and the OBJECT has a data length of 9.

Reply:
Fixed, please review. Thanks @SparkiDev